client/models.py

```diff
@@ -0,0 +1,83 @@
+
+
+class User(models.Model):
+    """An abstract class which holds basic user data:
+    Attributes:
+        user_name: a string which holds the user name, Also PK.
+        first_name: a string which holds the user first name.
+        last_name: a string which holds the user last name.
+        password: a string which holds the user password.
+        user_type: a string which holds the user type, either Supplier or Client.
+    """
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False,
+                                 validators=[MinLengthValidator(6)],
+                                 help_text="User Name")
+    first_name = models.CharField(max_length=32,
+                                  validators=[MinLengthValidator(1)],
+                                  help_text="First Name")
+    last_name = models.CharField(max_length=32,
+                                 validators=[MinLengthValidator(1)],
+                                 help_text="Last Name")
+    password = models.CharField(max_length=16,
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+    user_type = models.CharField(max_length=16,
+                                 validators=[MinLengthValidator(6)],
+                                 null=False,
+                                 choices=[('client', 'client'), ('supplier', 'supplier')],
+                                 help_text="User Type - supplier/client")
+
+    class Meta:
+        abstract = True
+
+
+class Client(User):
+    area = models.CharField(max_length=32,
+                            validators=[MinLengthValidator(1)],
+                            help_text="area")
+
+    @staticmethod
+    def get_all_clients():
+        """Fetches list of all Clients.
+        Args:
+            None.
+        Returns:
+            returns a queryset containing all the Clients.
+        """
+        return Client.objects.all()
+
+    @staticmethod
+    def save_client(client):
+        """Saves new client 'safely'
+        Args:
+            client - the Client wanted to be saved.
+        Returns:
+            None.
+        raises:
+        ValidationError error: if fields input aren't valid.
+        """
+        try:
+            Client.full_clean(client)
+            client.save()
+        except ValidationError as error:
+            raise error
+
+    @staticmethod
+    def delete_client(client):
+        """Deletes client 'safely'
+        Args:
+            client - the client wanted to be deleted
+        Returns:
+            None.
+        raises:
+        ObjectDoesNotExist error: if the client is not in DB.
+        """
+        if client not in Client.get_all_clients():
+            raise ObjectDoesNotExist
+        client.delete()
```

Comment:
You have a race condition here. What happens if the client is deleted between the time you check if it exists and the time it's deleted. It's better to avoid the check and instead wrap the delete in try/except and handle the exception raised if it does not exist.